core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java

```diff
@@ -345,14 +345,14 @@ public CompletableFuture<List<BlockStatus>> load(List<Block> blocks, UfsReadOpti
           .thenRun(() -> {
             try {
               blockWriter.close();
+              NioDirectBufferPool.release(buf);
             } catch (IOException e) {
               throw AlluxioRuntimeException.from(e);
-            } finally {
-              NioDirectBufferPool.release(buf);
             }
           })
           .thenRun(() -> commitBlock(sessionId, blockId, false))
```

Comment:
Seems like if we get exception at this block, then we would release the buf twice, which would cause a problem

Reply:
we can just put release after commitBlock, then we would be safe

Reply:
done